### DISCUSSION TITLE
Missing dsym in Firebase Crashlytics even though upload is successful

### DISCUSSION THREAD

**Discussion by thomasdf** [codemagic-ci-cd/codemagic-docs#2531] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

Run a build that includes this part in post build:

```
  echo "Find build artifacts"
  dsymPath=$(find $CM_BUILD_DIR/build/ios/archive/Runner.xcarchive -name "*.dSYM" | head -1)
  if [[ -z ${dsymPath} ]]
  then
    echo "No debug symbols were found, skip publishing to Firebase Crashlytics"
  else
    echo "Publishing debug symbols from $dsymPath to Firebase Crashlytics"
    ls -d -- ios/Pods/*
    $CM_BUILD_DIR/ios/Pods/FirebaseCrashlytics/upload-symbols \
      -gsp ios/Runner/GoogleService-Info.plist -p ios $dsymPath
  fi
```

### Expected results

Dsym files are uploaded to Firebase Crashlytics.

### Actual results

Output for that part of the build is:
```
Find build artifacts
Publishing debug symbols from /Users/builder/clone/build/ios/archive/Runner.xcarchive/dSYMs/in_app_review.framework.dSYM to Firebase Crashlytics
ios/Pods/Firebase
ios/Pods/FirebaseAnalytics
ios/Pods/FirebaseAppCheckInterop
ios/Pods/FirebaseAuth
ios/Pods/FirebaseCore
ios/Pods/FirebaseCoreExtension
ios/Pods/FirebaseCoreInternal
ios/Pods/FirebaseCrashlytics
ios/Pods/FirebaseInstallations
ios/Pods/FirebaseMessaging
ios/Pods/FirebaseRemoteConfigInterop
ios/Pods/FirebaseSessions
ios/Pods/GTMSessionFetcher
ios/Pods/GoogleAppMeasurement
ios/Pods/GoogleDataTransport
ios/Pods/GoogleUtilities
ios/Pods/Headers
ios/Pods/Local Podspecs
ios/Pods/Manifest.lock
ios/Pods/Mantle
ios/Pods/Pods.xcodeproj
ios/Pods/PromisesObjC
ios/Pods/PromisesSwift
ios/Pods/RecaptchaInterop
ios/Pods/SDWebImage
ios/Pods/SDWebImageWebPCoder
ios/Pods/Target Support Files
ios/Pods/libwebp
ios/Pods/nanopb
Successfully submitted symbols for architecture arm64 with UUID f4feecd3c66632698ad8dd543de36c0f in dSYM: /Users/builder/clone/build/ios/archive/Runner.xcarchive/dSYMs/in_app_review.framework.dSYM
Successfully uploaded Crashlytics symbols
```

But, Crashlytics still complain about missing dsym-files for these versions. It used to work before, but doesn't work anylonger.

### Build id (optional)

6639d6b891bea5d3aeef3dee

**Comment by dtrdic** [comment]
hey @thomasdf make sure you are adding this script in the post-publish section, not in the post-build section
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/38918425/8819f8bc-8495-4554-86eb-f2ea16f675e2)

try also by removing `| head -1`which is used to limit the output to the first result found by the find command.
```
echo "Find build artifacts"
  dsymPath=$(find $CM_BUILD_DIR/build/ios/archive/Runner.xcarchive -name "Runner.app.dSYM")
  if [[ -z ${dsymPath} ]]
  then
    echo "No debug symbols were found, skip publishing to Firebase Crashlytics"
  else
    echo "Publishing debug symbols from $dsymPath to Firebase Crashlytics"
    ls -d -- ios/Pods/*
    $CM_BUILD_DIR/ios/Pods/FirebaseCrashlytics/upload-symbols -gsp ios/Runner/GoogleService-Info.plist -p ios $dsymPath
  fi
```

**Comment by dtrdic** [comment]
I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new discussion.

**Comment by malikbrh** [comment]
Facing the same issue with dSYM missing in Firebase console.

It seems like the problem is with UUID matching, the script with the `| head -1` seems to always upload a single dSYM file with a UUID not matching the new version I'm building.

Here for example I'm hoping to see the `C723` imported but it always updates the one below not facing my new update.
<img width="587" height="81" alt="Screenshot 2025-08-25 at 11 46 26" src="https://github.com/user-attachments/assets/e2633190-e51b-479c-a233-cc2558fb2a32" />

When adapting the script without the head -1 it now picks up the files I need, the docs might need an update !
